### ISSUE TITLE
Unhandled check on request response

### ISSUE THREAD

**Issue by bizob2828** [SparkPost/node-sparkpost#21]
Naively checking a status code of an object and referencing another key without checking its parents

Patch below
```
diff --git a/lib/transmission.js b/lib/transmission.js
index befa640..e7e17e7 100644
--- a/lib/transmission.js
+++ b/lib/transmission.js
@@ -122,7 +122,7 @@ module.exports = {
     request.post(options, function(err, res, body) {
       if (err) {
         callback(err);
-      } else if (res.statusCode !== 200) {
+      } else if (res && res.body && res.statusCode !== 200) {
         callback(res.body.errors);
       } else {
         callback(null, body);
```

**Comment by jmartin4563** [comment]
Per discussion with Bob, the else-if statement needs to stay the same (otherwise we'll fall through to the successful else statement). We will do a second conditional inside the else-if to check for the existence of res and res.body and throw a generic error if either of them are undefined.

**Comment by jmartin4563** [comment]
Closed with pull request #22, publishing v1.0.5 to npm registry
